### PR TITLE
feat(node)!: Refactor errors and stop Syncer on fatal ones

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: node/src/store.rs [hunk 1]
```diff
@@ -202,14 +182,42 @@ pub enum StoreError {
...
+    #[error("Contraints not met: {0}")]
+    ContraintsNotMet(BlockRangesError),
+
+    // TODO: Same hash for two different heights is not really possible
+    // and `ExtendedHeader::validate` would return an error.
+    // Remove this when a type-safe validation is implemented.
+    /// Hash already exists in the store.
+    #[error("Hash {0} already exists in store")]
+    HashExists(Hash),
+}
```
Comment (oblique) [inline node/src/store.rs:208]: Should I merge it back to `StoreError`? When I created it I wanted to make it easier to identify non-fatal insertion errors.

[the remark this answers is not in the document]
Comment (zvolin) [reply node/src/store.rs:208]: no strong opinion on that honestly, both are fine

**Review by zvolin — APPROVED** [review]
nice, tho I'm a bit afraid that one day we'll forget that 'only fatal errors' must be propagated :see_no_evil: although I cannot think of anything better

File: node/src/store.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (oblique) [reply node/src/store.rs:208]: Ok let's keep it

**Pull request merged**